apps/meteor/client/components/dashboards/periods.ts

```diff
@@ -23,10 +23,22 @@ const lastNDays =
 		return { start, end };
 	};
 
-const getLast6Months = () => {
-	const last6Months = moment().subtract(6, 'months');
-	return moment().diff(last6Months, 'days');
-};
+export const getClosedPeriod =
+	(
+		startOf: 'year' | 'month' | 'week',
+		subtract = 0,
+	): ((utc: boolean) => {
+		start: Date;
+		end: Date;
+	}) =>
+	(utc): { start: Date; end: Date } => {
+		const date = utc ? new Date() : moment(new Date()).utc().toDate();
+
+		const start = moment(date).subtract(subtract, 'months').startOf(startOf).toDate();
+		const end = moment(date).endOf('day').toDate();
```

Comment:
I noticed two things:

- The ternary condition seems to be reversed. 
- Also it instantiates local time then shifts it to utc (`moment().utc()`). Not sure if this is the desired behavior, considering `lastNDays` returns utc+0 (`moment.utc()`) when `utc` flag is `true`
- It might be possible to reduce the number of conversions to `Date` and back to `moment` (not tested).

```suggestion
		const date = utc ? moment().utc().toDate() : new Date();

		const start = date.subtract(subtract, 'months').startOf(startOf).toDate();
		const end = date.endOf('day').toDate();
```

Reply:
> It might be possible to reduce the number of conversions to Date and back to moment (not tested).
The multiple `moment()` calls are to avoid isssues with [moment's mutability](https://momentjs.com/guides/#/lib-concepts/mutability/) , in your sugggestion the end value would be the end of the day set in the start variable

The ternary is indeed flipped, let me fix that